### DISCUSSION TITLE
ModuleNotFoundError: No module named 'cattrs.converters'

### DISCUSSION THREAD

**Discussion by byteinsight** [theOehrly/Fast-F1#323] [General]
Hi,

I'm trying to use FastF1 within a python/django venv I'm developing in Pycharm.

If I create a standard project in Python and import FastF1 I can run the simplest example no problem.

If I import FastF1 within the Django venv the error described is immediate.

_ModuleNotFoundError: No module named 'cattrs.converters'_

The modules are all present and of the same versions.    Are there any known issues with this approach?   It seems very odd.

Thanks. 

SOLUTION.   Packages must be installed via the File > Settings > Project > Python Interpreter and not via PIP and the terminal provided in the IDE.

**Comment by theOehrly** [comment]
I need some more clarification from you regarding this problem.

When you refer to a "django venv", do you mean a "normal" python venv? How do you create that venv?

What is "a standard project in Python" for you?

Regarding specifically PyCharm, your solution is odd. You should not need to go through PyCharms settings. When the correct virtual interpreter is selected, the associated venv should be active in PyCharms terminal and a normal pip install should work. If you use pip install in a system terminal outside of PyCharm, you need to activate the correct venv yourself.

**Reply by byteinsight** [reply]
Hi,  Thanks for responding.  It was indeed very odd.      

To create the Django venv I describe - in Pycharm > New Project > Django > Python Interpreter > New Environment using Virtual Env. 

Standard Project was just a simple Python app (still in Pycharm) but using the system wide interpreter which is an option available when following the command string above but  selecting alt option under Python Interpreter. 

"When the correct virtual interpreter is selected, the associated venv should be active in PyCharms terminal and a normal pip install should work"   This is what I believed as well and I'm sure its worked just fine in the past.  

Thanks again for checking in and kudos for the package.   Even though I have only scratched the surface its amazing.

**Reply by theOehrly** [reply]
I quickly tried to reproduce this now, but I couldn't. I tried it with installing FastF1 in the django venv as well as with inherited packages from the base environment. With both options everything just worked.
If you find a way to reliably reproduce this, then I can take another look at it. The way it is now, it is not really worth hunting for this fairly niche problem that might not even be caused by FastF1.
